types/utils.go

```diff
@@ -292,9 +305,15 @@ func GetRandomTx() Tx {
 }
 
 // GetRandomBytes returns a byte slice of random bytes of length n.
+// It uses crypto/rand for cryptographically secure random number generation.
 func GetRandomBytes(n int) []byte {
+	if n < 0 {
+		panic(fmt.Errorf("negative length %d is not allowed", n))
+	}
```

Comment:
We could simplify this by jus making `n` a `uint`. Then by definition it can't be negative.